buildkite/packages.go

```diff
@@ -85,3 +84,35 @@ func (ps *PackagesService) Create(ctx context.Context, organizationSlug, registr
 
 	return p, resp, err
 }
+
+// normalizeToFile takes and io.Reader (which might itself already be a file, but could be a stream or other source) and
+// writes it to a temporary file, returning the file handle.
+// The file is written to a temporary directory, and then renamed to the desired filename.
+// We do this normalization to ensure that we can accurately calculate the Content-Length of the request body, which is
+// required by S3. We write to disk (instead of buffering in memory) to avoid memory exhaustion for large files.
+func normalizeToFile(r io.Reader, filename string) (*os.File, error) {
+	basename := filepath.Base(filename)
+	f, err := os.CreateTemp("", basename)
+	if err != nil {
+		return nil, fmt.Errorf("creating temporary file: %v", err)
+	}
+
+	_, err = io.Copy(f, r)
+	if err != nil {
+		return nil, fmt.Errorf("writing to temporary file: %v", err)
+	}
+
+	_, err = f.Seek(0, 0)
```

Comment:
I prefer to using the constant for the second option `f.Seek(0, io.SeekStart)` as it makes it more approachable.

Reply:
yep, good call, will add